src/node/rpc/endpoint_registry.h

```diff
@@ -351,28 +392,43 @@ namespace ccf
        * @param v Boolean indicating whether the user identity must be known
        * @return This Endpoint for further modification
        */
+      CCF_DEPRECATED("Replace with add_authentication")
       Endpoint& set_require_client_identity(bool v)
       {
-        if (!v && registry != nullptr && !registry->has_certs())
-        {
-          LOG_INFO_FMT(
-            "Disabling client identity requirement on {} Endpoint has no "
-            "effect "
-            "since its registry does not have certificates table",
-            dispatch.uri_path);
-          return *this;
-        }
-
         properties.require_client_identity = v;
         return *this;
       }
 
+      CCF_DEPRECATED("Replace with add_authentication")
       Endpoint& set_require_jwt_authentication(bool v)
       {
         properties.require_jwt_authentication = v;
         return *this;
       }
 
+      /** Add an authentication policy which will be checked before executing
+       * this endpoint.
+       *
+       * When multiple policies are specified, any single successful check is
+       * sufficient to grant access, even if others fail. If all policies fail,
+       * the first can set an error status on the response, and the endpoint
+       * will not be invoked. If no policies are specified, then by default the
+       * endpoint accepts any request without an authentication check.
```

Comment:
I guess this makes sense for new apps to default to no authentication but for existing apps that rely on the current default user cert check, if they don't read the release notes carefully, upgrading CCF will silently open all their app endpoints to unauthenticated clients. 

Should we default to user cert check for the next release at least?

Reply:
That's a good question, I'd be tempted to say yes, but on the other hand it looks like the popular default is really going to be jwt, not user certs...

Reply:
After discussion with @jumaffre, I am going to make the following changes:

1. Add validation in install() to check at least one authentication policy is present, throw otherwise. We have not found a good way to do this at compile time, but if there is one it can always be done later.
2. Add an explicit empty_auth policy to all anonymous endpoints.

Apart from this being what seems right now, it's also a good starting point: if we relax it, user code won't break.

Reply:
Throwing in install() is very awkward: it cause the enclave_create_node ECALL to fail, and there's essentially no way to tell the user what went wrong. I am going to make this a log warning for now, and defer to #2032 finding a better enforcement mechanism.